File: example/src/main/java/jp/wasabeef/example/recyclerview/AdapterSampleActivity.java
package jp.wasabeef.example.recyclerview;

import android.content.Context;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.GridLayoutManager;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.animation.OvershootInterpolator;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.Spinner;
import java.util.ArrayList;
import java.util.Arrays;
import jp.wasabeef.recyclerview.adapters.AlphaInAnimationAdapter;
import jp.wasabeef.recyclerview.adapters.AnimationAdapter;
import jp.wasabeef.recyclerview.adapters.ScaleInAnimationAdapter;
import jp.wasabeef.recyclerview.adapters.SlideInBottomAnimationAdapter;
import jp.wasabeef.recyclerview.adapters.SlideInLeftAnimationAdapter;
import jp.wasabeef.recyclerview.adapters.SlideInRightAnimationAdapter;
import jp.wasabeef.recyclerview.animators.FadeInAnimator;

/**
 * Created by Wasabeef on 2015/01/03.
 */
public class AdapterSampleActivity extends AppCompatActivity {

  enum Type {
    AlphaIn {
      @Override public AnimationAdapter get(Context context) {
        MainAdapter adapter = new MainAdapter(context, new ArrayList<>(Arrays.asList(data)));
        return new AlphaInAnimationAdapter(adapter);
      }
    },
    ScaleIn {
      @Override public AnimationAdapter get(Context context) {
        MainAdapter adapter = new MainAdapter(context, new ArrayList<>(Arrays.asList(data)));
        return new ScaleInAnimationAdapter(adapter);
      }
    },
    SlideInBottom {
      @Override public AnimationAdapter get(Context context) {
        MainAdapter adapter = new MainAdapter(context, new ArrayList<>(Arrays.asList(data)));
        return new SlideInBottomAnimationAdapter(adapter);
      }
    },
    SlideInLeft {
      @Override public AnimationAdapter get(Context context) {
        MainAdapter adapter = new MainAdapter(context, new ArrayList<>(Arrays.asList(data)));
        return new SlideInLeftAnimationAdapter(adapter);
      }
    },
    SlideInRight {
      @Override public AnimationAdapter get(Context context) {
        MainAdapter adapter = new MainAdapter(context, new ArrayList<>(Arrays.asList(data)));
        return new SlideInRightAnimationAdapter(adapter);
      }
    };

    public abstract AnimationAdapter get(Context context);
  }

  private static String[] data = new String[] {
      "Apple", "Ball", "Camera", "Day", "Egg", "Foo", "Google", "Hello", "Iron", "Japan", "Coke",
      "Dog", "Cat", "Yahoo", "Sony", "Canon", "Fujitsu", "USA", "Nexus", "LINE", "Haskell", "C++",
      "Java", "Go", "Swift", "Objective-c", "Ruby", "PHP", "Bash", "ksh", "C", "Groovy", "Kotlin",
      "Chip", "Japan", "U.S.A", "San Francisco", "Paris", "Tokyo", "Silicon Valley", "London",
      "Spain", "China", "Taiwan", "Asia", "New York", "France", "Kyoto", "Android", "Google",
      "iPhone", "iPad", "iPod", "Wasabeef"
  };

  @Override protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_adapter_sample);

    Toolbar toolbar = (Toolbar) findViewById(R.id.tool_bar);
    setSupportActionBar(toolbar);
    getSupportActionBar().setDisplayShowTitleEnabled(false);

    final RecyclerView recyclerView = (RecyclerView) findViewById(R.id.list);

    if (getIntent().getBooleanExtra("GRID", true)) {
      recyclerView.setLayoutManager(new GridLayoutManager(this, 2));
    } else {
      recyclerView.setLayoutManager(new LinearLayoutManager(this));
    }

    Spinner spinner = (Spinner) findViewById(R.id.spinner);
    ArrayAdapter<String> spinnerAdapter =
        new ArrayAdapter<>(this, android.R.layout.simple_list_item_1);
    for (Type type : Type.values()) {
      spinnerAdapter.add(type.name());
    }
    spinner.setAdapter(spinnerAdapter);
    spinner.setOnItemSelectedListener(new AdapterView.OnItemSelectedListener() {
      @Override
      public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {
        AnimationAdapter adapter = Type.values()[position].get(AdapterSampleActivity.this);
        adapter.setFirstOnly(true);
        adapter.setDuration(500);
        adapter.setInterpolator(new OvershootInterpolator(.5f));
        recyclerView.setAdapter(adapter);
      }

      @Override public void onNothingSelected(AdapterView<?> parent) {
      }
    });

    recyclerView.setItemAnimator(new FadeInAnimator());
    MainAdapter adapter = new MainAdapter(this, new ArrayList<>(Arrays.asList(data)));
    AlphaInAnimationAdapter alphaAdapter = new AlphaInAnimationAdapter(adapter);
    alphaAdapter.setFirstOnly(true);
    alphaAdapter.setDuration(500);
    alphaAdapter.setInterpolator(new OvershootInterpolator(.5f));
    recyclerView.setAdapter(alphaAdapter);
  }
}

File: example/src/main/java/jp/wasabeef/example/recyclerview/AdapterSampleActivity.kt
package jp.wasabeef.example.recyclerview

import android.content.Context
import android.os.Bundle
import android.view.View
import android.view.animation.OvershootInterpolator
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.Spinner
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.GridLayoutManager
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import jp.wasabeef.recyclerview.adapters.AlphaInAnimationAdapter
import jp.wasabeef.recyclerview.adapters.AnimationAdapter
import jp.wasabeef.recyclerview.adapters.ScaleInAnimationAdapter
import jp.wasabeef.recyclerview.adapters.SlideInBottomAnimationAdapter
import jp.wasabeef.recyclerview.adapters.SlideInLeftAnimationAdapter
import jp.wasabeef.recyclerview.adapters.SlideInRightAnimationAdapter
import jp.wasabeef.recyclerview.animators.FadeInAnimator

/**
 * Created by Daichi Furiya / Wasabeef on 2020/08/26.
 */
class AdapterSampleActivity : AppCompatActivity() {

  internal enum class Type {
    AlphaIn {
      override operator fun get(context: Context): AnimationAdapter {
        return AlphaInAnimationAdapter(MainAdapter(context, SampleData.LIST.toMutableList()))
      }
    },
    ScaleIn {
      override operator fun get(context: Context): AnimationAdapter {
        return ScaleInAnimationAdapter(MainAdapter(context, SampleData.LIST.toMutableList()))
      }
    },
    SlideInBottom {
      override operator fun get(context: Context): AnimationAdapter {
        return SlideInBottomAnimationAdapter(MainAdapter(context, SampleData.LIST.toMutableList()))
      }
    },
    SlideInLeft {
      override operator fun get(context: Context): AnimationAdapter {
        return SlideInLeftAnimationAdapter(MainAdapter(context, SampleData.LIST.toMutableList()))
      }
    },
    SlideInRight {
      override operator fun get(context: Context): AnimationAdapter {
        return SlideInRightAnimationAdapter(MainAdapter(context, SampleData.LIST.toMutableList()))
      }
    };

    abstract operator fun get(context: Context): AnimationAdapter
  }

  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_adapter_sample)

    setSupportActionBar(findViewById(R.id.tool_bar))
    supportActionBar?.setDisplayShowTitleEnabled(false)

    val recyclerView = findViewById<RecyclerView>(R.id.list)
    recyclerView.layoutManager = if (intent.getBooleanExtra(MainActivity.KEY_GRID, true)) {
      GridLayoutManager(this, 2)
    } else {
      LinearLayoutManager(this)
    }

    val spinner = findViewById<Spinner>(R.id.spinner)
    spinner.adapter = ArrayAdapter<String>(this, android.R.layout.simple_list_item_1).apply {
      for (type in Type.values()) add(type.name)
    }
    spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
      override fun onItemSelected(parent: AdapterView<*>, view: View, position: Int, id: Long) {
        recyclerView.adapter = Type.values()[position][view.context].apply {
          setFirstOnly(true)
          setDuration(500)
          setInterpolator(OvershootInterpolator(.5f))
        }
      }

      override fun onNothingSelected(parent: AdapterView<*>) {
        // no-op
      }
    }

    recyclerView.itemAnimator = FadeInAnimator()
    val adapter = MainAdapter(this, SampleData.LIST.toMutableList())
    recyclerView.adapter = AlphaInAnimationAdapter(adapter).apply {
      setFirstOnly(true)
      setDuration(500)
      setInterpolator(OvershootInterpolator(.5f))
    }
  }
}
